python-sdk/src/astro/databases/snowflake.py

```diff
@@ -645,7 +645,8 @@ def load_pandas_dataframe_to_table(
         :param if_exists: Strategy to be used in case the target table already exists.
         :param chunk_size: Specify the number of rows in each batch to be written at a time.
         """
-
+        if source_dataframe.empty:
```

Comment:
Should this check also be added in the base class method? https://github.com/astronomer/astro-sdk/blob/0e9d07b53bb2218dabecdbb991dd49cc2ff94cfe/python-sdk/src/astro/databases/base.py#L532

Reply:
I'll not mind adding it in the base class too but some do DB allow to table without a column.

Reply:
In my opinion, the method is called `load_pandas_dataframe_to_table`. So in case the dataframe is empty, there is nothing to load, shouldn't we alert in that case?

Reply:
> I'll not mind adding it in the base class too but some DB do allow to table without a column.

Interesting. I am curious which db does that and why 🤔

Reply:
I don't know the exact reason but Postgres and bigquery do allow to creation of a table without a column.  You might like to read some more opinions here https://stackoverflow.com/questions/18024585/why-to-create-empty-no-rows-no-columns-table-in-postgresql

Reply:
I feel that we should have a separate method which should check this in the base class. That method can be overridden for Postgres and Bigquery.

Reply:
I have wrapped the error with the function `_assert_not_empty_df` in the base database and also add the check in base database too